simulator/main.rs

```diff
@@ -67,6 +116,343 @@ enum Value {
     Blob(Vec<u8>),
 }
 
+impl ArbitraryFrom<Vec<&Value>> for Value {
+    fn arbitrary_from<R: Rng>(rng: &mut R, t: &Vec<&Value>) -> Self {
+        if t.is_empty() {
+            return Value::Null;
+        }
+
+        let index = rng.gen_range(0..t.len());
+        t[index].clone()
+    }
+}
+
+impl ArbitraryFrom<ColumnType> for Value {
+    fn arbitrary_from<R: Rng>(rng: &mut R, t: &ColumnType) -> Self {
+        match t {
+            ColumnType::Integer => Value::Integer(rng.gen_range(i64::MIN..i64::MAX)),
+            ColumnType::Float => Value::Float(rng.gen_range(-1e10..1e10)),
+            ColumnType::Text => Value::Text(gen_random_text(rng)),
+            ColumnType::Blob => Value::Blob(gen_random_text(rng).as_bytes().to_vec()),
+        }
+    }
+}
+
+struct LTValue(Value);
+
+impl ArbitraryFrom<Vec<&Value>> for LTValue {
+    fn arbitrary_from<R: Rng>(rng: &mut R, t: &Vec<&Value>) -> Self {
+        if t.is_empty() {
+            return LTValue(Value::Null);
+        }
+
+        let index = rng.gen_range(0..t.len());
+        LTValue::arbitrary_from(rng, t[index])
+    }
+}
+
+impl ArbitraryFrom<Value> for LTValue {
+    fn arbitrary_from<R: Rng>(rng: &mut R, t: &Value) -> Self {
```

Comment:
Nit s/t/value

Reply:
Makes sense 👍🏻

Reply:
Changed all the one character names to longer and descriptive versions.